jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java

```diff
@@ -261,20 +255,29 @@ private static File getHomeDir() {
     private static void updateMapWithArgValue(Map<String, String> result, Map<String, String> args, String argString) {
         if (argString.contains("=") || argString.contains(":")) {
             String[] argStringParts = argString.split("[=:]");
-            String argStringValue = argString.substring(argStringParts[0].length() + 1);
+            String argStringKey = argStringParts[0];
+            String argStringValue = determineFinalArgValue(argString, argStringParts, args);
             if (argStringValue.startsWith("\"") || argStringValue.startsWith("'")) {
                 // Replaces surrounding quotes
-                argStringValue = trimSurroundingQuotes(argStringValue);
+                argStringValue = argStringValue.replaceAll("[\"']", "");
```

Comment:
This is not OK, you haven't ported the correct regex. It will fail for strings such as `"hello 'you'"` or `'hello "you"'`
Please, be very careful when dealing with regular expressions, they are the nightmare of every developer.